handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -288,6 +287,7 @@ protected void deallocate() {
                 PlatformDependent.throwException(cause);
             }
         }
+        leak = leakDetection ? leakDetector.track(this) : null;
```

Comment:
would be nice to add a comment explaining why this is at the end so we don't move it by accident later.